### PR TITLE
Add 'phdthesis' and 'mastersthesis' types for a preferred-citation.

### PR DESCRIPTION
**Pull request by ionathanch**
The corresponding BibTeX types are:
* https://www.bibtex.com/t/template-mastersthesis/
* https://www.bibtex.com/t/template-phdthesis/

And the corresponding APA format is from https://owl.purdue.edu/owl/research_and_citation/apa_style/apa_formatting_and_style_guide/reference_list_other_print_sources.html.

The `type` field is for specifying a particular type of thesis or dissertation, e.g. licentiate thesis, habilitation thesis, Bachelor's thesis, etc.

### CODE REVIEW

File: test/files/formatter/preferred-citation-masters-thesis-no-type.cff [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# A PhD thesis preferred citation
```
Comment (hainesr) [inline test/files/formatter/preferred-citation-masters-thesis-no-type.cff:1]: This comment doesn't match the title of the file or the rest of the contents. Please can you fix?

**Review by hainesr — CHANGES_REQUESTED** [review]
This looks great - thank you very much for the PR.

There's just one tiny change to a comment that I've marked, otherwise this looks good to go!

File: test/files/formatter/preferred-citation-masters-thesis-no-type.cff (hunk 1 above)
Comment (ionathanch) [reply test/files/formatter/preferred-citation-masters-thesis-no-type.cff]: Done!

**Review by hainesr — APPROVED** [review]

**Comment by hainesr** [comment]
Many thanks for this work @ionathanch 🚀

**Pull request merged**